### PR TITLE
fix: BFT quorum for period CAS

### PR DESCRIPTION
**Pull request by johannbarbie**
resolves https://github.com/leapdao/leap-contracts/issues/263

- [x] 1. add more extensive unit tests for CAS bitmap checks, covering different edge cases e.g. 0/3, 2/3, 3/3, 3/4. Both for failing and success scenarios.
- [x]  2. test the following scenario:
epoch length 3
slots taken: 0 and 2. Slot 1 is empty (validator logout)
submission should require only 2 sigs (2/2)
- [ ] 3. change neededSigs implementation for the tests to pass

3. has not been implemented, due to the following gas measurements:

```
period submission without CAS: 103578                                  98%
period submission with CAS at 3 epochLength: 105036					  100%
period submission with CAS at 255 epochLength: 144180				  137%	
period submission with CAS and active-slot-count at 3 EL:   111523    106%    
period submission with CAS and active-slot-count at 255 EL: 700615    667%
```
 checking slots (reading storage) is quite expensive, and increases with the number of slots.

### CODE REVIEW

File: test/poaOperator.js [hunk 1]
```diff
@@ -74,18 +74,89 @@ contract('PoaOperator', (accounts) => {
...
+        const consensusRoot = '0x03';
+        const casBitmap = '0xffffffffffffffffffffffffffffffffffffffffffc000000000000000000000'; // 170 bits set
+        await operator.submitPeriodWithCas(0, p[1], consensusRoot, casBitmap, { from: alice }).should.be.rejectedWith(EVMRevert);
+      });
+
+      it('should allow submission with CAS bitmap of 171/255', async () => {
+        const consensusRoot = '0x03';
+        const casBitmap = '0xffffffffffffffffffffffffffffffffffffffffffe000000000000000000000'; // 171 bits set
+        await operator.submitPeriodWithCas(0, p[1], consensusRoot, casBitmap, { from: alice }).should.be.fulfilled;
+        p[2] = await bridge.tipHash();
```
Comment (troggy) [inline test/poaOperator.js]: unnecessary — no used

**Comment by troggy** [comment]
- [ ] code review
- [ ] unit tests
- [ ] integration tests with 3 validator nodes
